src/Microsoft.ML.AutoML/Experiment/ModelContainer.cs

```diff
@@ -30,8 +31,10 @@ internal ModelContainer(MLContext mlContext, FileInfo fileInfo, ITransformer mod
             }
         }
 
-        public ITransformer GetModel()
+        public ITransformer GetModel(bool isModelDisposed)
         {
+            if (isModelDisposed)
+                throw new ObjectDisposedException("Trained model was disposed before");
             // If model stored in memory, return it
```

Comment:
Why is it necessary to throw this exception specifically? If the user tried to access a disposed object, won't the framework automatically throw this exception?

Reply:
There is no warning given when one attempts to use a disposed object, so throwing this exception is necessary.